swift/WorkflowUI/Sources/Screen/Screen.swift

```diff
@@ -19,4 +19,21 @@
 ///
 /// Conforming types contain any information needed to populate a screen: data,
 /// styling, event handlers, etc.
-public protocol Screen {}
+public protocol Screen {
+
+    /// The type of key for this screen.
+    associatedtype Key: Hashable = ObjectIdentifier
+
+
+    /// The key for this screen. This is often used by containers to determine screen equality.
+    /// The default value is the object identifier of the screen type.
+    var key: Key { get }
```

Comment:
The ergonomics of this are much better, but doesn't this create conflicts if two unrelated screens happen to provide the same `Key`? Having this on the container API means the scope of the key is limited to that container.